### ISSUE TITLE
Support for the New Box View APIs

### ISSUE THREAD

**Issue by akshaykumar6** [box/viewer.js#234]
I am using the [Box view APIs](https://view.box.com/reference#overview) to render the document which will be officially closed by January 2018. And I am unable to find any JS plugins for the [New Box View](https://developer.box.com/box-view) which has the same features as viewer.js. As I want to use the page events and plain HTML5 rendering for documents.

Are you planning to support the new Box APIs? Is there any other JS library which would provide the same features for new APIs?

Also, I came across [Box Javascript Preview SDK](https://docs.box.com/v2.0/docs/box-javascript-preview-sdk) which is rendering documents in iframe. Will this have the features of viewer.js in future?

**Comment by tonyjin** [comment]
We're focusing our development on the Javascript Preview SDK, which will power the New Box View. More features will be added as time goes on.

The biggest benefit of New Box View is that you will be able to preview all the file types that Box supports (media, 3d, images, text, documents, etc.), not just documents. The link you added (https://docs.box.com/docs/box-javascript-preview-sdk) is not loading the Javascript SDK in an iframe - check out the CodePen to see the exact code being run.

You'll be able to use the New Box View with either:
- The plain JS Preview SDK (which you linked)
- Hosted iframe embed via the Content API (https://docs.box.com/reference#get-embed-link)

**Comment by akshaykumar6** [comment]
Thanks a lot @tonyjin for such a quick response!!

I look forward to migrating to the New Box View. I actually want to use the page events, layout modes and transitions in my product. Will you be developing the same functionalities as [viewer.js](https://github.com/box/viewer.js) in the JS preview SDK?

**Comment by tonyjin** [comment]
Yup! Functional parity with viewer.js is on our backlog, but I don't have a timeline to share right now.

**Comment by MSchip** [comment]
@tonyjin Do you have even a rough timeline estimate? Will things like layouts and navigation be available before the January 2018 deprecation?
